Tests/Basic/OutputByteStreamTests.swift

```diff
@@ -25,30 +25,50 @@ class OutputByteStreamTests: XCTestCase {
         let streamable: Streamable = Character("!")
         stream.write(streamable)
 
-        stream.flush()
         
         XCTAssertEqual(stream.position, "Hello, world!".utf8.count)
+        stream.flush()
         XCTAssertEqual(stream.bytes, "Hello, world!")
     }
     
     func testStreamOperator() {
-        let stream = OutputByteStream()
+        let stream = BufferedOutputByteStream()
 
         let streamable: Streamable = Character("!")
         stream <<< "Hello" <<< Character(",") <<< Character(" ") <<< [UInt8]("wor".utf8) <<< [UInt8]("world".utf8)[3..<5] <<< streamable
         
         XCTAssertEqual(stream.position, "Hello, world!".utf8.count)
         XCTAssertEqual(stream.bytes, "Hello, world!")
 
-        let stream2 = OutputByteStream()
+        let stream2 = BufferedOutputByteStream()
         stream2 <<< (0..<5)
         XCTAssertEqual(stream2.bytes, [0, 1, 2, 3, 4])
     }
     
+    func testBufferCorrectness() {
+        let smallBlock = [UInt8](repeating: 2, count: 1 << 14)
+        // Check small / big interleavings for various block sizes (to prove buffer transfer optimizations).
+        for blockSize in [1 << 10, 1 << 12, 1 << 13, 1 << 14] {
+            let bigBlock = [UInt8](repeating: 1, count: blockSize)
```

Comment:
I think this should get even more interesting data than this to catch cases where we copy a small piece but with the wrong offset, but can be fixed up later...